File: Python/linked_list02.py
class Node:
    def __init__(self, dat=0):
        self.data = dat
        self.next : Node = None
    
def create():
    d = int(input("Enter Data: "))
    return Node(d)
def insert(start, datt, pos='first', valid_pos=0):
    new = Node(datt)
    if pos == 'first':
        new.next = start
        return new
    elif pos == 'last':
        if start is None:
            return new
        temp = start
        while temp.next is not None:
            temp = temp.next
        temp.next = new
        return start
    elif pos == 'any':
        if start is None:
            return new
        temp = start
        i = 1
        while i < valid_pos-1 and temp.next is not None:
            temp = temp.next
            i += 1
        new.next = temp.next
        temp.next = new
        return start  
    else:
        print("Invalid position declaration")
        return start
    
def delete(start, pos='first', valid_pos=0):
    if pos == 'first':
        return start.next
    elif pos == 'last':
        temp = start
        while temp.next.next is not None:
            temp = temp.next
        temp.next = None
        return start
    elif pos == 'any':
        temp = start
        n = 1
        while n < valid_pos-1 and temp.next.next is not None:
            temp = temp.next
            n += 1
        temp.next = temp.next.next
        return start
    else:
        print("Invalid position declaration")
        return start

def disp(start):
    print()
    print("Start", end=" -> ")
    while start:
        print(start.data, end=" -> ")
        start = start.next
    print("End")
def main():
    start : Node = None
    new : Node = None
    ch = 'y'
    while(ch == 'y'):
        new = create()
        if start is None:
            start = new
        else:
            temp : Node = start
            while(temp.next is not None):
                temp = temp.next
            temp.next = new
        ch = input('Do you want more node? [y/n]')
    start = insert(start, 100, )
    start = insert(start, 300)
    start = insert(start, 200, 3)
    disp(start)    

if __name__ == "__main__":
    main()
    
#insert at first and at last

File: CPP/linkedList02.cpp
#include<stdio.h>
#include<stdlib.h>
#include<string>
#include<conio.h>

using namespace std;

typedef struct Node
{
    int data;
    struct Node *next = NULL;
}NODE;

class LinkedList
{
    private:
        NODE *start;
    public:
        LinkedList()
        {
            this->start = NULL;
        }
        void create()
        {
            char ch = 'y';
            NODE *new_node = NULL;
            NODE *temp = NULL;
            do
            {
                new_node = new Node();
                new_node->next = NULL;
                printf("\nEnter Data: ");
                scanf("%d", &new_node->data);
                if(this->start == NULL)
                {
                    this->start = new_node;
                }
                else
                {
                    temp = start;
                    while(temp->next != NULL)
                    {
                        temp = temp->next;
                    }
                    temp->next = new_node;
                }
                printf("\nDo you want more data [y/n]?...");
                scanf(" %c", &ch);
            } while (ch == 'y');
        }
        void disp()
        {
            NODE *temp = NULL;
            if(this->start == NULL)
            {
                printf("\nNo elements found");
            }
            temp = start;
            printf("\nSTART -> ");
            while(temp != NULL)
            {
                printf("%d -> ", temp->data);
                temp = temp->next;
            }
            printf("END");
        }
        void add(int data)
        {
            NODE *new_node = new Node();
            new_node->data = data;
            new_node->next = NULL;
            if(start == NULL)
            {
                start = new_node;
            }
            else
            {
                NODE *temp = start;
                while (temp->next != NULL)
                {
                    temp = temp->next;
                }
                temp->next = new_node;
            }
        }
        void insert(int data, string pos, int index)
        {
            NODE *new_node = new Node();
            new_node->next = NULL;
            new_node->data = data;   
            if (pos == "first")
            {
                if (this->start == NULL)
                {
                    this->start = new_node;
                }
                else
                {
                    new_node->next = this->start;
                    this->start = new_node;
                }
            }
            else if (pos == "last")
            {
                if (this->start == NULL)
                {
                    this->start = new_node;
                }
                else
                {
                    NODE *temp = start;
                    while (temp->next != NULL)
                    {
                        temp = temp->next;
                    }
                    temp->next = new_node;
                }
            }
            else if (pos == "any")
            {
                if (this->start == NULL)
                {
                    this->start = new_node;
                }
                else
                {
                    NODE *temp = start;
                    int i=0;
                    while (i < index-1)
                    {
                        temp = temp->next;
                        i++;
                    }
                    new_node->next = temp->next;
                    temp->next = new_node;
                }
            }
            else
            {
                printf("Invalid Position");
            }
        }
        void count()
        {
            Node *temp = start;
            int c = 0;
            while(temp != NULL)
            {
                temp = temp->next;
                c++;
            }
            printf("Number of elements: %d", c);
        }
};
int main()
{
    LinkedList p;
    p.create();
    p.add(10);
    p.insert(100, "first", 0);
    p.insert(300, "last", 0);
    p.insert(150, "any", 2);
    p.count();
    p.disp();
    return 0;
}